### PR TITLE
python homework class 4

### PR DESCRIPTION
**Pull request by IgorOlshanskiy**
python homework class 4 - DONE

### CODE REVIEW

File: homework_class4.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+# that, given an array A of N integers, returns the smallest positive integer (greater than 0) that does not occur in A.
+# For example, given A = [1, 3, 6, 4, 1, 2], the function should return 5.
+# Given A = [1, 2, 3], the function should return 4.
+# Given A = [−1, −3], the function should return 1.
+# Assume that:
+# N is an integer within the range [1..100,000];
+# each element of array A is an integer within the range [−1,000,000..1,000,000].
+
+
+def task_1(a_tup):
```
Comment (mmykhailenko) [inline homework_class4.py]: easy way

    for i in range(1, len(l) + 2):
        if i not in l:
            return i

File: homework_class4.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+# N is an integer within the range [1..2,147,483,647].
+# ============================
+
+def task_2(num):
+    bin_str = str(bin(num))[2:]
+    bin_gap = False
+    bin_max = 0
+    bin_cnt = 0
+    for i in bin_str:
+        if i == '1':
```
Comment (mmykhailenko) [inline homework_class4.py]: you could use `split` instead `if` then all code will be easy

formatted_n = '{0:b}'.format(n)
split_formatted_n = formatted_n.split('1')

return len(max(split_formatted_n, key=len))

File: homework_class4.py (hunk 1 above)
Comment (IgorOlshanskiy) [reply homework_class4.py]: Got it

File: homework_class4.py (hunk 2 above)
Comment (IgorOlshanskiy) [reply homework_class4.py]: got it

**Pull request merged**